src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -400,6 +401,14 @@ private void CheckSnapshot(ITextSnapshot snapshot)
                 var newTagTrees = ImmutableDictionary<ITextBuffer, TagSpanIntervalTree<TTag>>.Empty;
                 foreach (var buffer in buffers)
                 {
+                    if (!buffersToTag.Contains(buffer))
```

Comment:
Shouldn't we be updating the computation of buffers in line 392 or so instead? It seems like you can just delete that and make this loop over buffersToTag instead?

(I'm guessing the `spansTagged.Select(dss => dss.SnapshotSpan.Snapshot.TextBuffer)` is the equivalent of your `buffersToTag` hence this seems we're adding stuff only to filter it back out again.)

Reply:
Jeez, i hate like 382 :).  Why not just require a realized list...

Reply:
➡️ This was substantially simplified in b9e81c978d7847af4f29e831064585b1345e4d97. Rationale for the individual changes is detailed in the commit message.